src/Fixer/PhpBasic/Feature/VisibilityPropertiesFixer.php

```diff
@@ -293,26 +292,53 @@ private function getPropertyVariable(Tokens $tokens, int $key, bool $propertyExc
         return null;
     }
 
-    private function insertVariablePropertyWarning(Tokens $tokens, int $key, string $convention)
+    private function insertVariablePropertyWarning(Tokens $tokens, int $key, string $convention): void
     {
         $this->insertComment(
             $tokens,
-            $tokens->getNextTokenOfKind($key, [';']),
+            $key,
             $tokens[$key]->getContent(),
             $convention,
         );
     }
 
-    private function insertComment(Tokens $tokens, int $insertIndex, string $propertyName, string $convention)
+    private function insertComment(Tokens $tokens, int $insertIndex, string $propertyName, string $convention): void
     {
         $comment = '// TODO: "' . $propertyName . '" - ' . $convention;
-        if (!$tokens[$tokens->getNextNonWhitespace($insertIndex)]->isGivenKind(T_COMMENT)) {
+
+        do {
+            $token = $tokens[$insertIndex++];
+        } while (!substr_count($token->getContent(), "\n"));
+
+        if (!$tokens[$tokens->getNextNonWhitespace($insertIndex - 3)]->isGivenKind(T_COMMENT)) {
             $tokens->insertSlices([
-                ++$insertIndex => [
+                $insertIndex - 1 => [
                     new Token([T_WHITESPACE, ' ']),
                     new Token([T_COMMENT, $comment]),
                 ],
             ]);
         }
     }
+
+    /**
+     * @param array<Token> $previousTokens
+     * @return Token|null
```

Comment:
that one is redundand, info is available in method signature

Reply:
I can't agree with you. In the method signature we have array $previousTokens, but it is not clear what is in the array. And we have the following code:
.....
foreach ($previousTokens as $previousToken) {
            if ($previousToken->isGivenKind(
.......

Without this clarification (array<Token>) IDE doesn't know that $previousToken is a Token (https://github.com/paysera/php-style-guide?tab=readme-ov-file#phpdoc-on-arrays).

@return Token|null in the code because of this rule "If we use PhpDoc comment, it must contain all information about parameters, return type and exceptions that the method throws" (https://github.com/paysera/php-style-guide?tab=readme-ov-file#phpdoc-contents).

Reply:
I have changed array<Token> to Token[].
TypeHintingArgumentsFixer has problems with this construction "array<Token>". I will fix this in the next PR.

Reply:
I only meant `Token|null` part, not full PHPdoc.